src/feed_rss.js

```diff
@@ -109,12 +102,14 @@ $(document).ready(function(){
     });
 
     /* search articles after keywords in title, category & teaser */
-    $("input[type='text']").on('keypress keyup', function () {
+    $("input[type='text']").on('keyup', function () {
         let value = $(this).val().toLowerCase();
 
         $("article").filter(function(){
             $(this).toggle($(this).text().toLowerCase().indexOf(value) > -1)
         });
-        
+        $(".box1").filter(function(){
```

Comment:
To summarize this: you're buildung, sorting and rendering your model to html .. but when searching you parse your own presentation instead of using your model again? Why? search is business functionality, not presentation functionality.